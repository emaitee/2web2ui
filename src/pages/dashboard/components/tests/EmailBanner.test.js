import { shallow } from 'enzyme';
import React from 'react';

import EmailBanner from '../EmailBanner';

describe('Component: Email Banner', () => {
  const props = {
    sendingStatus: 'not-sending',
    handleResend: jest.fn(() => true)
  };

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<EmailBanner {...props} />);
  });

  it('should show default email banner', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should show sending sending content', () => {
    wrapper.setProps({ sendingStatus: 'sending' });
    expect(wrapper).toMatchSnapshot();
  });

  it('should show sent content', () => {
    wrapper.setProps({ sendingStatus: 'sent' });
    expect(wrapper).toMatchSnapshot();
  });

});
