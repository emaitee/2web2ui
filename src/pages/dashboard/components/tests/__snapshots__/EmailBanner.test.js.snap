// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Email Banner should show default email banner 1`] = `
<Banner
  action={
    Object {
      "content": "Resend Email",
      "onClick": [MockFunction],
    }
  }
  status="info"
  title="Verify your email address"
>
  <p>
    Please click the link in the email we sent you to verify your email address and unlock higher daily sending limits.
  </p>
</Banner>
`;

exports[`Component: Email Banner should show sending sending content 1`] = `
<Banner
  action={
    Object {
      "content": "Sending..",
      "disabled": true,
    }
  }
  status="info"
  title="Verify your email address"
>
  <p>
    Please click the link in the email we sent you to verify your email address and unlock higher daily sending limits.
  </p>
</Banner>
`;

exports[`Component: Email Banner should show sent content 1`] = `
<Banner
  action={null}
  status="info"
  title="Verify your email address"
>
  <p>
    Please click the link in the email we sent you to verify your email address and unlock higher daily sending limits.
  </p>
  <p>
    <strong>
      Email sent!
    </strong>
  </p>
</Banner>
`;
